toucan_connectors/databricks/databricks_connector.py

```
@@ -0,0 +1,131 @@
+
+    def get_status(self) -> ConnectorStatus:
+        # Check hostname
+        try:
+            self.check_hostname(self.host)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(0, False), error=str(e))
+        # Check port
+        try:
+            self.check_port(self.host, self.port)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(1, False), error=str(e))
+        try:
+            pyodbc.connect(self._build_connection_string())
+        except pyodbc.InterfaceError as e:
+            details = self._get_details(3, False)
+            details[2] = ('Connected to Databricks', True)
+            return ConnectorStatus(status=False, details=details, error=e.args[0])
+        except pyodbc.Error as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(2, False), error=e.args[0]
+            )
+        return ConnectorStatus(status=True, details=self._get_details(3, True), error=None)
+
+    def check_cluster_running(self) -> Dict[str, str]:
+        endpoint = f'https://{self.host}/api/2.0/clusters/get'
+        headers = {'login': 'token', 'password': self.pwd.get_secret_value()}
+        data = {'cluster_id': self.http_path.split('/')[-1]}
+        return requests.get(endpoint, headers=headers, json=data).json().get('state') == 'RUNNING'
+
+    def start_cluster(self) -> None:
+        endpoint = f'https://{self.host}/api/2.0/clusters/start'
+        headers = {'login': 'token', 'password': self.pwd.get_secret_value()}
+        data = {'cluster_id': self.http_path.split('/')[-1]}
+        resp = requests.post(endpoint, headers=headers, json=data)
+        if resp.status_code == 200:
+            logger.info('Databricks cluster started')
+        else:
+            logger.info(resp.json().get('message', 'Failed to start Databricks cluster'))
```

It could be nice for the caller to have the information that the cluster did not start. Maybe raise an exception here ?

---

yes, I add that.